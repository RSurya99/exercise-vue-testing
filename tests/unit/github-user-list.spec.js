import GithubUserList from '@/components/exercise/github-user-list.vue'
import { shallowMount } from '@vue/test-utils'
import flushPromises from 'flush-promises'

describe('Github User List', () => {
  it('add new card when form username is filled and Add button clicked', async () => {
    const wrapper = shallowMount(GithubUserList)
    const input = wrapper.find('[data-testid=username-input]')
    const btn = wrapper.find('[data-testid=btn-submit]')

    input.setValue('rsurya99')
    const jsonMock = jest.fn().mockResolvedValue({
      login: 'RSurya99',
    })
    window.fetch = jest.fn().mockResolvedValue({
      json: jsonMock,
    })
    await btn.trigger('submit')
    await flushPromises()
    expect(window.fetch).toHaveBeenCalledWith('https://api.github.com/users/rsurya99')
    expect(wrapper.vm.data[0]).toEqual({
      login: 'RSurya99',
    })
  })

  it('clear all list of github user card when Clear All button clicked', async () => {
    const wrapper = shallowMount(GithubUserList)
    const btn = wrapper.find('[data-testid=btn-clearall]')

    await btn.trigger('click')
    expect(wrapper.vm.data).toEqual([])
  })
})
