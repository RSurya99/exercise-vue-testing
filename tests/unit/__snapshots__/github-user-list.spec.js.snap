// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github User List renders correctly 1`] = `
<div style="text-align: left; width: 640px;">
  <header style="display: flex; justify-content: space-between;">
    <form class="form"><label for="username"><input id="username" type="text" data-testid="username-input"></label> <button type="submit" data-testid="btn-submit">Add</button></form> <button type="button" data-testid="btn-clearall">Clear All</button>
  </header>
  <div></div>
</div>
`;
